src/Card.java

```diff
@@ -1,60 +1,44 @@
+import java.util.Arrays;
 import java.util.HashMap;
+import java.util.List;
+import java.util.stream.IntStream;
 
 public class Card {
 
-    String cardCode;
+    private final String cardCode;
+    private final String cardColor;
+    private final String cardValue;
+    private Integer intCardValue;
+    private final HashMap<String, Integer> alphabeticCardValue;
+    private final List<String> colors = Arrays.asList("S","C","D","H");
+    private final List<String> letterValues = Arrays.asList("J","Q","K","A");
 
-    public Card(String cardCode) throws IllegalArgumentException {
-        this.cardCode = cardCode;
+    public Card(String cardCode) {
+        this.cardCode = cardCode.toUpperCase();
+        cardColor = this.cardCode.substring(0, 1);
+        cardValue = this.cardCode.substring(1);
+        alphabeticCardValue = new HashMap<>();
+        IntStream.range(0,4).forEach(i -> alphabeticCardValue.put(letterValues.get(i),i+11));
 
-        String cardColor = this.cardCode.substring(0, 1).toUpperCase();
 
-        if (!cardColor.contentEquals("S") &&
-            !cardColor.contentEquals("C") &&
-            !cardColor.contentEquals("D") &&
-            !cardColor.contentEquals("H"))
+        if (!colors.contains(cardColor))
         {
             throw new IllegalArgumentException("card color isn't valid: " + cardColor);
         }
 
-        String cardValue = cardCode.substring(1).toUpperCase();
-        Integer intCardValue;
-
-        HashMap<String, Integer> alphabeticCardValue = new HashMap<>();
-        alphabeticCardValue.put("J", 11);
-        alphabeticCardValue.put("Q", 12);
-        alphabeticCardValue.put("K", 13);
-        alphabeticCardValue.put("A", 14);
-
-        if (alphabeticCardValue.get(cardCode.substring(1).toUpperCase()) == null) {
+        if (!alphabeticCardValue.containsKey(cardValue)) {
             // raises exception if cardValue is a letter, but not J/Q/K/A
             intCardValue = Integer.parseInt(cardValue);
-            if (intCardValue > 10) {
-                throw new IllegalArgumentException("card number isn't valid: " + intCardValue);
-            }
-            if (intCardValue < 2) {
+            if (intCardValue > 10 || intCardValue < 2) {
```

Comment:
Ezekre erdemes hasznalni a Guava preconditions library-jet:
https://github.com/google/guava/wiki/PreconditionsExplained

A Guava egy csomo hasznos dolgot tartalmazo java lib, a google csinalta. Hozza tudod adni a jar-t a classpath-hoz, vagy ha mavent hasznalsz akkor a pom-hoz is hozzaadhatod:
https://github.com/google/guava

Guava-val ez a csunyasag:
```
if (intCardValue > 10) {
                 throw new IllegalArgumentException("card number isn't valid: " + intCardValue);
}
```
Ezze alakithato: `checkArgument(intCardValue >= 2 && intCardValue <= 10, "card number isn't valid: " + intCardValue)`